runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -540,28 +527,11 @@ def up(self, verbose: bool = True, force: bool = False):
 
         if self.launcher_type == LauncherType.DEN:
             logger.info("Launching cluster with Den.")
-            launcher = DenLauncher(cluster=self, force=force)
-            data = launcher.up(verbose=verbose)
-            self._update_from_den_response(data)
+            DenLauncher.up(cluster=self, verbose=verbose, force=force)
 
         else:
             logger.info("Launching cluster locally.")
-            launcher = LocalLauncher(cluster=self)
-            launcher.up(verbose=verbose)
-            self._update_from_sky_status()
-
-            if self.domain:
-                logger.info(
-                    f"Cluster has been launched with the custom domain '{self.domain}'. "
-                    "Please add an A record to your DNS provider to point this domain to the cluster's "
-                    f"public IP address ({self.address}) to ensure successful requests."
-                )
-
-            # TODO: Maybe apply the below to both cases?
-            self.restart_server()
-
-            if rns_client.autosave_resources():
-                self.save()
+            LocalLauncher.up(cluster=self, verbose=verbose)
 
         return self
```

Comment:
does this approach still ensure that "self" here is updated from the `launcher.up()` call?

Reply:
Yup! As long as the cluster reference is maintained in the method, the changes pass through to `self`!
Took a bit of experimentation but definitely working